proxy/server.js

```diff
@@ -1,10 +1,94 @@
-const express = require('express')
-const path = require('path');
-const app = express();
+require('newrelic');
+const fs = require('fs');
+const http = require('http');
+const axios = require('axios');
+const { redisClient } = require('./redisindex');
+
 const port = process.env.PORT || 8000;
 
-app.use(express.static(path.join(__dirname, 'public')));
+// app.use(express.static(path.join(__dirname, './public')));
+
+const clientBundles = './public/services';
+const serverBundles = './templates/services';
+const serviceConfig = require('./service-config.json');
+const services = require('./loader.js')(clientBundles, serverBundles, serviceConfig);
+
+// services = {
+//   Reviews: require('path to file server bundle ').default,
+//   Photo: require('path to photo server bundle').default,
+// }
 
-app.listen(port, () => {
-  console.log(`server running at: http://localhost:${port}`)
+const React = require('react');
+const ReactDom = require('react-dom/server');
+const Layout = require('./templates/layout');
+const App = require('./templates/app');
+const Scripts = require('./templates/scripts');
+
+const renderComponents = (components, props = {}) => Object.keys(components).map((item) => {
+  const component = React.createElement(components[item], props);
+  return ReactDom.renderToString(component);
+  // '<Reviews />'
 });
+
+http.createServer((req, res) => {
+  if (req.url.startsWith('/restaurants')) {
+    const id = Number(req.url.slice(13));
+    const redisKey = `Restaurant${id}`;
+    redisClient.get(redisKey, (err, reply) => {
+      if (err) {
+        res.writeHead(404);
+        res.write('Redis', err);
+        res.end();
+      } else if (reply !== null) {
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(reply);
+      } else {
```

Comment:
I think you could simplify your logic in this function by removing lines 38 through 41. Because if there is no data or no key, then Redis will error or return undefined so instead of having to do something with the error, you could simply start with if (reply !== null) , then 'else' axios call to service